src/Symfony/Component/Validator/Constraints/ExpressionValidator.php

```diff
@@ -65,13 +68,87 @@ public function validate($value, Constraint $constraint)
 
     private function getExpressionLanguage()
     {
-        if (null === $this->expressionLanguage) {
+        if (!$this->expressionLanguage instanceof ExpressionLanguage) {
             if (!class_exists('Symfony\Component\ExpressionLanguage\ExpressionLanguage')) {
                 throw new LogicException('Unable to use expressions as the Symfony ExpressionLanguage component is not installed.');
             }
             $this->expressionLanguage = new ExpressionLanguage();
+
+            $this->addIsValidFunction();
         }
 
         return $this->expressionLanguage;
     }
+
+    private function addIsValidFunction(): void
+    {
+        if ($this->expressionLanguage->hasFunction('is_valid')) {
+            return;
+        }
+
+        $this->expressionLanguage->register('is_valid', function () {
```

Comment:
This code must be moved to a dedicated class `ExpressionLanguageProvider` by implementing `ExpressionFunctionProviderInterface` and later `$this->expressionLanguage->registerProvider(...)`.